### PR TITLE
[improve][cli] Pulsar shell, Pulsar admin, Pulsar client and Pulsar perf: support for Windows

### LINKED ISSUE #16688: Windows support for admin CLI tools
Currently for Windows users there's no easy way to run Pulsar CLI tools such as `pulsar-admin`, `pulsar-shell`, and so on.

The goal would be to give the possibility to a Windows user to download the Pulsar tarball and administrate a remote Pulsar cluster from their own Windows machine. 

Since the Pulsar Admin is a java application, the only implementation needed would be to:
* Create a .cmd file for pulsar-admin (`bin/pulsar-admin.cmd`)
* Replicate the same logic from the bash files (actually `bin/pulsar-admin-common.sh`)

This will become more useful starting from Pulsar 2.11 which will contains the Pulsar Shell artifact.

### PR DESCRIPTION
**Pull request by nicoloboschi**
Fixes #16688

TL; DR - now Pulsar shell works on Windows
Screen Shot 2022-08-23 at 6 02 50 PM

### Motivation
Currently pulsar shell is only runnable on Unix system because the entry point script is written in bash.

### Modifications

1. Provided a new `pulsar-shell.cmd`, `pulsar-admin.cmd`, `pulsar-client.cmd` and `pulsar-perf.cmd` files (written in Windows Command Prompt language) that starts their processes

It implements the exact same configurations from the current pulsar-shell, even the java8 detection. The only thing not implemented is the on-the-fly compile. Actually, the batch script only work if you unzip the distro tarball. Running it from the codebase (so without the `lib` dir) is not supported (and it's okay since currently it's not even possible develop Pulsar on a Win machine)

2. Added the jna lib to make the jline command prompt display colors correctly - will need to upgrade to the latest JNA version to fix some current issues

3. Added zip distribution to pulsar-shell (distribution/shell). The idea is to distribute both and suggest to download the .zip for Windows user

- [x] `doc-not-needed`

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -211,7 +211,7 @@ flexible messaging model and an intuitive client API.</description>
     <jakarta.activation.version>1.2.2</jakarta.activation.version>
     <jakarta.xml.bind.version>2.3.3</jakarta.xml.bind.version>
     <jakarta.validation.version>2.0.2</jakarta.validation.version>
-    <jna.version>4.2.0</jna.version>
+    <jna.version>5.12.1</jna.version>
```
Comment (eolivelli) [inline pom.xml:214]: I would move this change to a separate PR

File: pulsar-client-tools/src/main/java/org/apache/pulsar/shell/PulsarShell.java [hunk 2]
```diff
@@ -216,7 +216,6 @@ public void reload(Properties properties) throws Exception {
     }
 
     public void run() throws Exception {
-        System.setProperty("org.jline.terminal.dumb", "true");
```
Comment (eolivelli) [inline pulsar-client-tools/src/main/java/org/apache/pulsar/shell/PulsarShell.java:219]: what's the impact on Linux and Mac ?

**Review by eolivelli — CHANGES_REQUESTED** [review]

**Review by eolivelli — COMMENTED** [review]
https://github.com/apache/pulsar/issues/16688 refers also to pulsar-admin.
this patch does not completely fixes that issue.

Also we should do this also for pulsar-admin, pulsar-client, pulsar-perf and pulsar standalone

File: pulsar-client-tools/src/main/java/org/apache/pulsar/shell/PulsarShell.java (hunk 2 above)
Comment (nicoloboschi) [reply pulsar-client-tools/src/main/java/org/apache/pulsar/shell/PulsarShell.java:219]: Actually nothing. I had firstly misunderstood the meaning of this property. It's not useful

**Review by eolivelli — APPROVED** [review]
LGTM

great work !

**Comment by eolivelli** [comment]
@eolivelli license check is failing

File: pom.xml [hunk 3]
```diff
@@ -211,7 +211,7 @@ flexible messaging model and an intuitive client API.</description>
     <jakarta.activation.version>1.2.2</jakarta.activation.version>
     <jakarta.xml.bind.version>2.3.3</jakarta.xml.bind.version>
     <jakarta.validation.version>2.0.2</jakarta.validation.version>
-    <jna.version>5.12.1</jna.version>
+    <jna.version>4.2.0</jna.version>
```
Comment (tisonkun) [inline pom.xml:214]: I guess you'd prefer revert this change as described in https://github.com/apache/pulsar/pull/17262?

Comment (nicoloboschi) [reply pom.xml:214]: oh thanks, that's what I call a bad git rebase

File: distribution/shell/src/assemble/shell.xml [hunk 4]
```diff
@@ -24,6 +24,7 @@
   <id>shell</id>
   <formats>
     <format>tar.gz</format>
+    <format>zip</format>
```
[the remark this answers is not in the document]
Comment (nicoloboschi) [reply distribution/shell/src/assemble/shell.xml:27]: that's a good point. Pulsar shell currently is 50mb so I guess is not a big deal

**Review by tisonkun — APPROVED** [review]
I don't have a Windows environment to verify these scripts, though.

File: bin/pulsar-admin-common.cmd [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+if "%PULSAR_ROUTING_APPENDER_DEFAULT%" == "" set "PULSAR_ROUTING_APPENDER_DEFAULT=Console"
+
+set "OPTS=%OPTS% -Dpulsar.log.appender=%PULSAR_LOG_APPENDER%"
+set "OPTS=%OPTS% -Dpulsar.log.dir=%PULSAR_LOG_DIR%"
+set "OPTS=%OPTS% -Dpulsar.log.level=%PULSAR_LOG_LEVEL%"
+set "OPTS=%OPTS% -Dpulsar.log.root.level=%PULSAR_LOG_ROOT_LEVEL%"
+set "OPTS=%OPTS% -Dpulsar.routing.appender.default=%PULSAR_ROUTING_APPENDER_DEFAULT%"
+
+:error
+exit /b 1
```
Comment (BewareMyPower) [inline bin/pulsar-admin-common.cmd:82]: Why did you add this line? I cannot use `pulsar-shell` on Windows unless I removed this line.

Comment (nicoloboschi) [reply bin/pulsar-admin-common.cmd:82]: it's a leftover, we can drop it
